### PR TITLE
fix: SIGSEGV on parsing of config file. 

### PR DESCRIPTION
**Pull request by hassec**
This reverts a bug that was introduced by the `clang-tidy` change in #1659. (fyi @neheb)  

I'm just opening this already to make people aware of it. 
Before merging, I'd like to add a test that uses a config file, as it's a bit of a shame we hadn't noticed this.

### CODE REVIEW

**Comment by neheb** [comment]
Makes sense.

**Comment by kevinbackhouse** [comment]
Interesting. I guess this is because `INIReader::ValueHandler` tries to add elements to `reader->_values` before construction is complete. I don't much like that `void *user` stuff!

**Comment by hassec** [comment]
@kevinbackhouse yep that's it.   
I don't like it either but it's an external library that is designed with C in mind and the C++ part really is just a thin wrapper around it.

But on top of that, I'm not really a fan of having this library just dropped into our source like this and then even applying clang-tidy to it.      
We need a cleaner way to handle 3rd party libs, so I added it to #1515 

One could consider replacing it with something else but let's leave that discussion to #1515

**Comment by kevinbackhouse** [comment]
@hassec: Oh, I see. All of this code, including that `INIReader::ValueHandler` stuff, is copied from https://github.com/benhoyt/inih.
It looks like it's available as a [library on Ubuntu](https://packages.ubuntu.com/hirsute/libinih1), but I don't know what the situation is on other platforms.

**Comment by kevinbackhouse** [comment]
I think I would prefer if the test didn't write a file into the user's home directory. One suggestion is to get `getExiv2ConfigPath` to try `getenv("HOME")` first, so that the test can use the environment to tell it where to find the config file. Another suggestion is to a new command-line argument, like `--config=...`.

**Comment by hassec** [comment]
I kind of understand that point. But the test is set up in a way such that it can't overwrite a preexisting file, and it removes the file it creates at the end.
So there really isn't an observable impact for the user.

`--config` is an option I actually also thought about, but decided against it because I didn't want to add an extra option just to make the testing easier. 
The config file itself is already a bit of a corner case feature that is only used in a few maker notes for lens name overwriting.  

if you have strong feelings against the current solution, I'd say the `--config` flag option is the best one if we want to gain test coverage over this feature.

**Comment by kevinbackhouse** [comment]
The main reason why I don't like it is that it could make Exiv2 look like a malicious repo. People are quite worried about supply-chain attacks these days, so a build or test script that modifies files outside its own directory could raise suspicions.

**Comment by kevinbackhouse** [comment]
Another option that might be relatively quick to implement would be to check the current directory for a file named `.exiv2` first, before looking in the home directory.

**Comment by hassec** [comment]
@kevinbackhouse I've implemented the current working directory version because it seemed more straight forward.         


Turns out adding the option to the exiv2 app isn't hard but then one has to think about how to properly pass those forward to the library part of exiv2, and then also expose this setting to the library users. That would have been more work so I went with your suggestion :wink: 

Also, thanks for reviewing!! :+1:

File: man/man1/exiv2.1 [hunk 1]
```diff
@@ -682,7 +682,7 @@ the YYYY:MM:DD HH:MM:SS standard.
 .sp 1
 .ne 40
 .SH CONFIGURATION FILE
-Exiv2 can read an optional configuration file ~/.exiv2 on Unix systems and %USERPROFILE%\\exiv2.ini on Windows (using a Visual Studio build).  Cygwin and MinGW/msys2 follow the unix convention and use ~/.exiv2  You can determine the location of the configuration file with the command:
```
Comment (hassec) [inline man/man1/exiv2.1:685]: 2 things I'm not clear on:
1. Not sure why this specifically mentions "Visual Studio build"
2. it says MinGW and the others use a unix convention, but that's not what was implemented in the code... I've no clue what would make more sense for these platforms... 

@kevinbackhouse @alexvanderberkel @piponazo any suggestions?

Comment (kevinbackhouse) [reply man/man1/exiv2.1:685]: I don't know the history of this. I would remove "(using a Visual Studio build)" as it doesn't seem to add anything.

**Review by kevinbackhouse — DISMISSED** [review]

File: man/man1/exiv2.1 (hunk 1 above)
Comment (hassec) [reply man/man1/exiv2.1:685]: I went ahead and changed the man to reflect what is actually done in the code.   

@kevinbackhouse  that of course dismissed your review again :sweat_smile: So could you just approve again, and I'll go ahead and finally merge this :)

**Review by kevinbackhouse — APPROVED** [review]

**Pull request merged**